pkg/skaffold/build/cache/hash.go

```diff
@@ -38,15 +38,82 @@ import (
 
 // For testing
 var (
-	hashFunction           = cacheHasher
-	artifactConfigFunction = artifactConfig
+	newArtifactHasherFunc = newArtifactHasher
+	fileHasherFunc        = fileHasher
+	artifactConfigFunc    = artifactConfig
 )
 
-func getHashForArtifact(ctx context.Context, depLister DependencyLister, a *latest.Artifact, mode config.RunMode) (string, error) {
+type artifactHasher interface {
+	hash(ctx context.Context, a *latest.Artifact) (string, error)
+}
+
+// Artifacts is a map of [artifact image : artifact definition]
+type Artifacts map[string]*latest.Artifact
+
+// ToArtifacts creates an instance of `Artifacts` from `[]*latest.Artifact`
+func ToArtifacts(artifacts []*latest.Artifact) Artifacts {
```

Comment:
this is the most overloaded term in skaffold I think :) `ToArtifactMap`?

Reply:
renamed to `ArtifactGraph`